[DOC] Describe what bids-matlab can and cannot do (#84)
bids-standard/bids-matlab: README.md

### What this toolbox can do

- read the layout of a BIDS dataset (see `bids.layout`),
- perform queries on that layout to get information about the subjects, sessions, runs, modalities, metadata... contained within that dataset (see `bids.query`),
- generate a human readable report of the content of BIDS data set containing anatomical MRI, functional MRI, diffusion weighted imaging, field map data (see `bids.report`)
- read and write JSON files (see `bids.util.jsondecode` and `bids.util.jsonencode`) provided that the right [dependencies](#reading-and-writing-json-files) are installed,
- read and write TSV files (see `bids.util.tsvread` and `bids.util.tsvwrite`),

The behavior of this toolbox assumes that it is interacting with a valid BIDS dataset that should have been validated using [BIDS-validator](https://bids-standard.github.io/bids-validator/). Any unvalidated components may produce undefined behavior. Although, if you're BIDS-y enough, the behavior may be predictable.

### What this toolbox cannot do... yet

- parse the layout of "BIDS-derivative compatible" datasets (like fMRIprep),
- create BIDS compatible filenames or folder structures for raw or derivatives datasets,

> - basic copying of files to help with derivative dataset generation,

Reviewer: ```suggestion
- do basic copying of files to help with derivative dataset generation,
```
This should probably be a verb phrase, to satisfy parallel construction of the list items.

See also [PyBIDS](https://github.com/bids-standard/pybids) for Python and the [BIDS Starter Kit](https://github.com/bids-standard/bids-starter-kit).

## Features

### What this toolbox can do

- read the layout of a BIDS dataset (see `bids.layout`),
- perform queries on that layout to get information about the subjects, sessions, runs, modalities, metadata... contained within that dataset (see `bids.query`),
- generate a human readable report of the content of BIDS data set containing anatomical MRI, functional MRI, diffusion weighted imaging, field map data (see `bids.report`)
- read and write JSON files (see `bids.util.jsondecode` and `bids.util.jsonencode`) provided that the right [dependencies](#reading-and-writing-json-files) are installed,
- read and write TSV files (see `bids.util.tsvread` and `bids.util.tsvwrite`),
- parse typical BIDS filenames organized in key/value pairs (like this `key1-value_key2-value`) (see `bids.util.parse_filename`).

It also comes with some basic file name handling capabilities "inherited" from `spm_file` and `spm_select` (see `bids.util.file_utils`).

> The behavior of this toolbox assumes that it is interacting with a valid BIDS dataset that should have been validated using [BIDS-validator](https://bids-standard.github.io/bids-validator/). Any unvalidated components may produce undefined behavior. Although, if you're BIDS-y enough, the behavior may be predictable.

Author: Ah true !! Let me update that.

> The behavior of this toolbox assumes that it is interacting with a valid BIDS dataset that should have been validated using [BIDS-validator](https://bids-standard.github.io/bids-validator/). Any unvalidated components may produce undefined behavior. Although, if you're BIDS-y enough, the behavior may be predictable.

Author: ```suggestion
The behavior of this toolbox assumes that it is interacting with a valid BIDS dataset that should have been validated using [BIDS-validator](https://bids-standard.github.io/bids-validator/). If the Node.js version of the validator is [installed on your computer](https://github.com/bids-standard/bids-validator#quickstart), you can call it from the matlab prompt using `bids.validate`. Just be aware that any unvalidated components may produce undefined behavior. Although, if you're BIDS-y enough, the behavior may be predictable.
```

### What this toolbox cannot do... yet

> - parse the layout of "BIDS-derivative compatible" datasets (like fMRIprep),

Author: Ah! I meant "like those generated by fMRIprep" and not "like pyBIDS can". 

Which one is more helpful ? I tend to think people will know more about fMRIprep and would be more looking for an easy way to query them in Octave/Matlab rather than knowing that pyBIDS can parse them. 

Should we explain both?

> - parse the layout of "BIDS-derivative compatible" datasets (like fMRIprep),

Author: ```suggestion
- parse the layout of "BIDS-derivative compatible" datasets (like those generated by fMRIprep),
```

## Requirements

BIDS-MATLAB works with: 

- Octave 5.2.0 or newer
- MATLAB R2014a or newer  

> We aim for compatibility with the latest stable release of Octave at any time. Compatibility can sometimes also be achieved with older versions of Octave but this is not guaranteed.

Reviewer: ```suggestion
We aim for compatibility with the latest stable major version of Octave at any time. Compatibility can sometimes also be achieved with older versions of Octave but this is not guaranteed.
```

Maybe you should declare support for a full major version series like Octave 5.x instead of just the "latest stable release", which could be interpreted as "exactly 5.2.3" or similar.

Maybe declare a policy about how long Octave major versions are supported? For example, Octave 6.x is right around the corner. How long will you support Octave 5.x once 6.x is out?

And what about a policy for how long older Matlab versions are supported? If you support R2014a forever, you're eventually going to have a _lot_ of Matlab versions to test against, and you'll never get to adopt newer Matlab features.

## Requirements

BIDS-MATLAB works with: 

> - Octave 4.2.2 or newer

Author: ```suggestion
- Octave 5.2.0 or newer
```

- MATLAB R2014a or newer  

> In general we will try to use the latest version of Octave as a baseline for compatibility.

Reviewer 2: "We aim for compatibility with the latest stable release of Octave at any time. Compatibility can sometimes also be achieved with older versions of Octave but this is not guaranteed."

Author: ```suggestion
We aim for compatibility with the latest stable release of Octave at any time. Compatibility can sometimes also be achieved with older versions of Octave but this is not guaranteed.
```

> It may also work with older versions, but those are not actively supported.

Author: ```suggestion
```
